libsolidity/analysis/ControlFlowRevertPruner.h

```diff
@@ -0,0 +1,149 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTVisitor.h>
+#include <libsolidity/analysis/ControlFlowGraph.h>
+
+#include <libsolutil/Algorithms.h>
+
+namespace solidity::frontend
+{
+
+/// Analyses all function flows and removes any CFG nodes that make function
```

Comment:
Nodes or edges?

Reply:
Well, technically, the edges to the exit nodes of the reverting CFG node. I also add a new edge to the revert node..